pkg/service/gcs/gcs_client.go

```diff
@@ -60,6 +61,27 @@ type defaultGcsClient struct {
 	projectClient *resourcemanager.ProjectsClient
 }
 
+func (c *defaultGcsClient) DeleteObjectWithPrefix(ctxIn context.Context, bucket string, objectPrefixName string) error {
+	ctx, span := trace.StartSpan(ctxIn, "(*defaultGcsClient).DeleteObjectWithPrefix")
+	defer span.End()
+
+	objectIterator := c.client.Bucket(bucket).Objects(ctx, &storage.Query{Prefix: objectPrefixName})
+	for {
+		objAttrs, err := objectIterator.Next()
```

Comment:
What happens, if the list of objects is so long, that the http call will timeout, is a background job maybe better solution?